### ISSUE TITLE
Problems with installation

### ISSUE THREAD

**Issue by fmiotello** [PyLops/curvelops#20]
Hi everyone!
I'm having some problems installing `curvelops`. I'm following the [installation guide](https://pylops.github.io/curvelops/installation.html) but get this error during the execution of `python3 -m pip install git+https://github.com/PyLops/curvelops@0.23`:

```
>       error: command '/usr/bin/g++' failed with exit code 1
>       [end of output]
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel for curvelops
> Failed to build curvelops
> ERROR: Could not build wheels for curvelops, which is required to install pyproject.toml-based projects
```
 All the variables are set correctly, as specified in the guide.
I'm using `python 3.8` on `Ubuntu 22.04.1 LTS` and working on a `Conda` environment.

Can anyone help me on this?

**Comment by cako** [comment]
Hi @fmiotello, thanks for the issue.

Let's try to pinpoint where the error is coming from. The error is stating that the library could not be built due to a compiler error. This can have many causes. The first step is to grab the actual error that caused the compilation issue. I would suggest you log the entire output of the pip install command with:

```bash
FFTW=/path/to/fftw-2.1.5 FDCT=/path/to/CurveLab-2.1.3 python3 -m pip install git+https://github.com/PyLops/curvelops@0.23 --log curvelops.log
```

And link curvelab.log to the discussion (e.g. via pastebin or gist). That will give us some idea of what is going on. Second step is to make sure that everything is installed and ok. Please report the output of the following commands:

```bash
which gcc
gcc --version
which g++
g++ --version
echo $FFTW
echo $FDCT
```

Finally, I would encourage you to check your installation of FFTW 2.1.5 and CurveLab:
```bash
cd /path/to/fftw-2.1.5
make check

cd /path/to/CurveLab-2.1.3
make test
```

The FFTW should show in different lines:

```
--------------------------------------------------------------
         FFTW complex-complex transforms passed tests!
--------------------------------------------------------------

--------------------------------------------------------------
         RFFTW real-complex transforms passed tests!
--------------------------------------------------------------
```

The CurveLab testing should show something like this:

```bash
fdct3d_param 0 seconds
fdct3d_forward 2 seconds
fdct3d_inverse 1 seconds
max error 1.22468e-15
```

**Comment by fmiotello** [comment]
Hi @cako, thanks for your help!
I followed all the steps you indicated. [Here is the log](https://gist.github.com/fmiotello/08761e4d6e8e64abd30ade1970023366) of the installation command. 

These are the outputs of the commands you asked for:
```
> $ which gcc
/usr/bin/gcc

> $ gcc --version
gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

> $ which g++
/usr/bin/g++

> $ g++ --version
g++ (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

> $ echo $FFTW
/nas/home/fmiotello/dip-rir/opt/fftw-2.1.5

> $ echo $FDCT
/nas/home/fmiotello/dip-rir/opt/CurveLab-2.1.3
```

Finally, both the installation tests prompted the lines you indicated.

**Comment by cako** [comment]
Ok, so this is quite encouraging. Basically what is happening is that the library is not being found.

https://gist.github.com/fmiotello/08761e4d6e8e64abd30ade1970023366#file-curvelops-log-L194

So before we proceed, let's make sure the library is compiled. I realized that I didn't actually add the instructions for compilation in the docs, so that is patched:

https://pylops.github.io/curvelops/installation.html

Basically just run: 
```bash
make
make install
```
in the FFTW folder and
```bash
make lib
```
in the CurveLab folder.

**Comment by fmiotello** [comment]
Hi @cako! Thank you for your help, I finally managed to install `curvelops`. The problem was that the libraries were not compiled, as you pointed out.

Moreover, using
```
python3 -m pip install git+https://github.com/PyLops/curvelops@0.23
```
created some problems. The installation was successful but I wasn't able to import the library on my script. In my case, the solution was to install `curvelops` using
```
git clone https://github.com/PyLops/curvelops.git
cd curvelops
python3 -m pip install -e .
```
as pointed out in https://github.com/PyLops/curvelops/issues/18#issuecomment-1484012587.

**Comment by cako** [comment]
Awesome, enjoy!
